src/core/core.scale.js

```diff
@@ -916,28 +932,27 @@ module.exports = Element.extend({
 
 		if (gridLines.drawBorder) {
 			// Draw the line at the edge of the axis
-			context.lineWidth = helpers.valueAtIndexOrDefault(gridLines.lineWidth, 0);
-			context.strokeStyle = helpers.valueAtIndexOrDefault(gridLines.color, 0);
-			var x1 = me.left;
-			var x2 = me.right + axisWidth;
-			var y1 = me.top;
-			var y2 = me.bottom + axisWidth;
-
-			var aliasPixel = helpers.aliasPixel(context.lineWidth);
+			var firstLineWidth = axisWidth;
+			var lastLineWidth = helpers.valueAtIndexOrDefault(gridLines.lineWidth, ticks.length - 1, 0);
+			var x1 = helpers.alignLinePixel(chart, me.left, firstLineWidth) - firstLineWidth / 2;
+			var x2 = helpers.alignLinePixel(chart, me.right, lastLineWidth) + lastLineWidth / 2;
+			var y1 = helpers.alignLinePixel(chart, me.top, firstLineWidth) - firstLineWidth / 2;
+			var y2 = helpers.alignLinePixel(chart, me.bottom, lastLineWidth) + lastLineWidth / 2;
+
 			if (isHorizontal) {
-				y1 = y2 = options.position === 'top' ? me.bottom : me.top;
-				y1 += aliasPixel;
-				y2 += aliasPixel;
+				y1 = y2 = position === 'top' ? alignedBottom : alignedTop;
 			} else {
-				x1 = x2 = options.position === 'left' ? me.right : me.left;
-				x1 += aliasPixel;
-				x2 += aliasPixel;
+				x1 = x2 = position === 'left' ? alignedRight : alignedLeft;
 			}
 
-			context.beginPath();
-			context.moveTo(x1, y1);
-			context.lineTo(x2, y2);
-			context.stroke();
+			if (axisWidth) {
```

Comment:
Good catch! do we also need to compute all the code above if at the end we draw nothing? or should we break earlier, for example at line 933 (`if (gridLines.drawBorder && axisWidth)`)?

Reply:
I realized `axisWidth` has to be set to `0` if `gridLines.drawBorder` is `false`, so I moved this logic at the beginning of `draw`.